### ISSUE TITLE
Eager only loads related many to many relation for first result

### ISSUE THREAD

**Issue by akramali86** [Vincit/objection.js#389]
Hi, i'm having an issue where i'm trying to load a many to many relation using an eager query. But its only loading them for the first result.

This is the query which i'm trying to execute

```javascript
module.exports = function (filter = {}) {

    return Task
        .query()
        .where(filter)
        .eager('assignees(select)', {select: builder => builder.select(['id', 'firstName', 'lastName'])});

};
```

Relation Mappings for the Task model
```javascript
relationMappings: {
        client: {
            relation: Model.BelongsToOneRelation,
            modelClass: path.resolve(__dirname,'../models/client'),
            join: {
                from: 'task.clientId',
                to: 'client.id'
            }
        },
        project: {
            relation: Model.BelongsToOneRelation,
            modelClass: path.resolve(__dirname,'../models/project'),
            join: {
                from: 'task.projectId',
                to: 'project.id'
            }
        },      
        team: {
            relation: Model.BelongsToOneRelation,
            modelClass: path.resolve(__dirname,'../models/team'),
            join: {
                from: 'task.teamId',
                to: 'team.id'
            }
        },
        assignees: {
            relation: Model.ManyToManyRelation,
            modelClass: path.resolve(__dirname,'../models/user'),
            join: {
                from: 'task.id',
                through: {
                    from: 'task_user.taskId',
                    to: 'task_user.userId',
                    modelClass: path.resolve(__dirname,'../models/task_user')
                },
                to: 'user.id'
            }
        },
        createdBy: {
            relation: Model.BelongsToOneRelation,
            modelClass: path.resolve(__dirname,'../models/user'),
            join: {
                from: 'task.userId',
                to: 'user.id'
            }
        }
    }
```

The result of the above query is this:
```javascript
[
  {
    "id": 3,
    "title": "est est qui",
    "description": "minima illum nemo deleniti praesentium",
    "duration": 213,
    "link": "",
    "completedAt": null,
    "isComplete": false,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "ccd2de1a-a9e3-4bf0-be27-a42d0d40a9f3",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": [
      {
        "id": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
        "firstName": "akram",
        "lastName": "ali"
      },
      {
        "id": "22697a01-11c4-47fd-8d40-2da167befd45",
        "firstName": "Clifton",
        "lastName": "Cremin"
      },
      {
        "id": "a52640d3-08cd-41bf-8fd8-4d419b5a3a4f",
        "firstName": "Mallie",
        "lastName": "Koelpin"
      },
      {
        "id": "9b49f378-75e0-43fa-bb69-eeaa9e61304d",
        "firstName": "Matilde",
        "lastName": "Brown"
      },
      {
        "id": "49313dbf-dcbd-4725-a68e-c502e6aff0b9",
        "firstName": "Florine",
        "lastName": "Padberg"
      },
      {
        "id": "ccd2de1a-a9e3-4bf0-be27-a42d0d40a9f3",
        "firstName": "Desiree",
        "lastName": "Tromp"
      }
    ]
  },
  {
    "id": 11,
    "title": "accusamus ut modi",
    "description": "dolorem aliquid quos impedit asperiores",
    "duration": 147,
    "link": "",
    "completedAt": null,
    "isComplete": false,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "ccd2de1a-a9e3-4bf0-be27-a42d0d40a9f3",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 12,
    "title": "praesentium quia aliquid",
    "description": "aut eveniet qui dolor iusto",
    "duration": 31,
    "link": "",
    "completedAt": null,
    "isComplete": false,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "9b49f378-75e0-43fa-bb69-eeaa9e61304d",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 14,
    "title": "omnis iste harum",
    "description": "eaque natus eaque et culpa",
    "duration": 43,
    "link": "",
    "completedAt": null,
    "isComplete": false,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 17,
    "title": "voluptas fuga natus",
    "description": "voluptatibus magnam tempore ducimus neque",
    "duration": 158,
    "link": "",
    "completedAt": null,
    "isComplete": false,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "a52640d3-08cd-41bf-8fd8-4d419b5a3a4f",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 30,
    "title": "atque consequatur quia",
    "description": "ut deserunt non sequi voluptatum",
    "duration": 108,
    "link": "",
    "completedAt": "2017-05-27T18:45:32.388Z",
    "isComplete": true,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "49313dbf-dcbd-4725-a68e-c502e6aff0b9",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 31,
    "title": "pariatur mollitia tenetur",
    "description": "enim beatae veritatis reprehenderit debitis",
    "duration": 222,
    "link": "",
    "completedAt": "2017-05-27T18:45:32.388Z",
    "isComplete": true,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "49313dbf-dcbd-4725-a68e-c502e6aff0b9",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 44,
    "title": "nam autem nulla",
    "description": "eligendi veniam unde omnis nisi",
    "duration": 125,
    "link": "",
    "completedAt": "2017-05-27T18:45:32.388Z",
    "isComplete": true,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "49313dbf-dcbd-4725-a68e-c502e6aff0b9",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 45,
    "title": "est quia et",
    "description": "non enim recusandae voluptas quod",
    "duration": 137,
    "link": "",
    "completedAt": "2017-05-27T18:45:32.388Z",
    "isComplete": true,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  },
  {
    "id": 46,
    "title": "et debitis non",
    "description": "impedit recusandae modi minima asperiores",
    "duration": 132,
    "link": "",
    "completedAt": "2017-05-27T18:45:32.388Z",
    "isComplete": true,
    "teamId": 1,
    "clientId": 2,
    "projectId": null,
    "userId": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
    "createdAt": "2017-05-27T18:45:32.397Z",
    "updatedAt": "2017-05-27T18:45:32.397Z",
    "assignees": []
  }
]
```

As you can see, only the first result has its relations loaded, in the database every Task has at least 1 many to many relation with a user. 

This can be shown by fetching one of the above models, that appear to have no related models, using a first() operator.

```javascript
module.exports = function(filter = {}) {

    return Task
        .query()
        .where(filter)
        .eager('assignees(select)', {select: builder => builder.select(['id', 'firstName', 'lastName'])})
        .first();

};
```
This is the result of fetching task with the id of 46, which above shows has an empty assignees array

```javascript
{
  "id": 46,
  "title": "et debitis non",
  "description": "impedit recusandae modi minima asperiores",
  "duration": 132,
  "link": "",
  "completedAt": "2017-05-27T18:45:32.388Z",
  "isComplete": true,
  "teamId": 1,
  "clientId": 2,
  "projectId": null,
  "userId": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
  "createdAt": "2017-05-27T18:45:32.397Z",
  "updatedAt": "2017-05-27T18:45:32.397Z",
  "assignees": [
    {
      "id": "fb4a6def-57b5-4da8-8ca2-c0639cb0aecd",
      "firstName": "akram",
      "lastName": "ali"
    },
    {
      "id": "22697a01-11c4-47fd-8d40-2da167befd45",
      "firstName": "Clifton",
      "lastName": "Cremin"
    },
    {
      "id": "a52640d3-08cd-41bf-8fd8-4d419b5a3a4f",
      "firstName": "Mallie",
      "lastName": "Koelpin"
    },
    {
      "id": "9b49f378-75e0-43fa-bb69-eeaa9e61304d",
      "firstName": "Matilde",
      "lastName": "Brown"
    },
    {
      "id": "49313dbf-dcbd-4725-a68e-c502e6aff0b9",
      "firstName": "Florine",
      "lastName": "Padberg"
    },
    {
      "id": "ccd2de1a-a9e3-4bf0-be27-a42d0d40a9f3",
      "firstName": "Desiree",
      "lastName": "Tromp"
    }
  ]
}
```

I'm not sure if its an issue on my end with the query, or possibly a bug. I'm using Postgres, with objection on 0.7.12. The algorithm for the eager operator, is the default algorithm. Thanks for your help

**Comment by koskimas** [comment]
Sorry, I'm unabled to reproduce this. Could you create a minimal standalone app that I can easily run that reproduces the problem? I tried to create a similar query, and got all results (also there are multiple similar queries in the test set).

**Comment by akramali86** [comment]
Hi, Thanks for the quick response, After a lot of testing i figured out It appeared to be something to do with the user model having a default join algorithm. Removing the default join algorithm would make it run fine. I created a small app as you requested and to my surprise it was working as intended. I then uninstalled objection, and installed the latest version and it's been working perfectly. So i'm guessing something must have changed recently that fixed the issue. Thanks for your help and the this great library. I will close this issue as everything appears to be working as intended.
